src/EnergyPlus/BranchNodeConnections.cc

```diff
@@ -665,6 +665,7 @@ namespace BranchNodeConnections {
 					}
 				}
 				if ( ! IsValid ) {
+					if (NodeConnections(Loop1).ObjectType == "COILSYSTEM:INTEGRATEDHEATPUMP:AIRSOURCE") continue;
```

Comment:
The purpose of compsets is to validate parent-child object connections.  It has two main functions:
1.  To catch cases where the parent object uses node names that don't match the child object.  e.g. The Branch object refers to a chiller with Inlet Node X and Outlet Node Y but the chiller has no nodes by those names.
2.  To catch cases where the same object is used in more than one place.  e.g. Chilled Water Coil 1 is used in two different fan coil units.  Normally, this is illegal.  But for transpired solar collectors and this critter CoilSystem:IntegratedHeatPump:AirSource, it *is* legal for multiple parents to reference the same child.  In the example file for this, CoilSystem:IntegratedHeatPump:AirSource "ASIHP1" is referenced by a WaterHeater:HeatPump:PumpedCondenser, and an AirLoopHVAC:UnitaryHeatPump:AirToAir (twice).  So we need a general approach to allow components like this.  Let me think this through and offer a solution.